cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/provisioner/TetheringConf.java

```diff
@@ -38,10 +40,11 @@ private TetheringConf(String tetheredInstanceName, String tetheredNamespace) {
   /**
    * Create the conf from a property map while also performing validation.
    */
-  public static TetheringConf fromProperties(Map<String, String> properties) {
+  public static TetheringConf fromProperties(Map<String, String> properties, TetheringStore tetheringStore) {
```

Comment:
I think we should make validate a method that takes a PeerInfo object rather than forcing a call to be made when constructing this object. Then call it only in the validateProperties() method.

Forcing an extra I/O here adds more ways to fail (if TetheringStore is down) and potentially forces an extra call if the caller happened to have a PeerInfo around.

Reply:
fixed